### DISCUSSION TITLE
Should the LSP be a subcommand on the main `typst` CLI

### DISCUSSION THREAD

**Discussion by cmoog** [nvarner/typst-lsp#65] [General]
I'm wondering if there is a strong case for moving this into the `typst` CLI itself. Prior art for this pattern can be found in the [deno](https://github.com/denoland/deno/tree/main/cli/lsp) CLI, which embeds its own LSP. This prevents the possibility of version drift between the embedded LSP compiler and a separately installed, standalone compiler.

**Comment by beeb** [comment]
At the moment, version drift is prevent by doing the opposite, whereby the compiler is embedded into the LSP binary. But I would see this as an improvement in the case where the user would use the typst binary to compile and only use the LSP to check for diagnostic errors.
